pkg/imageregistry/imageregistry.go

```diff
@@ -0,0 +1,233 @@
+	}
+
+	err := builder.apiClient.Update(context.TODO(), builder.Definition)
+	if err == nil {
+		builder.Object = builder.Definition
+	}
+
+	return builder, err
+}
+
+// GetManagementState fetches imageRegistry ManagementState.
+func (builder *Builder) GetManagementState() (v1.ManagementState, error) {
+	if valid, err := builder.validate(); !valid {
+		return "", err
+	}
+
+	glog.V(100).Infof("Getting imageRegistry ManagementState configuration")
+
+	if !builder.Exists() {
+		return "", fmt.Errorf("imageRegistry object doesn't exist")
+	}
+
+	return builder.Object.Spec.ManagementState, nil
+}
+
+// WithManagementState returns true when succeeded to change imageRegistry operator management state.
+func (builder *Builder) WithManagementState(expectedManagementState v1.ManagementState) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Setting imageRegistry %s with expectedManagementState: %v",
+		builder.Definition.Name, expectedManagementState)
+
+	builder.Definition.Spec.ManagementState = expectedManagementState
+
+	return builder
+}
+
+// SetManagementState returns true when succeeded to change imageRegistry operator management state.
+func (builder *Builder) SetManagementState(expectedManagementState v1.ManagementState) error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Set imageRegistry %s ManagementState to the %v",
+		builder.Definition.Name, expectedManagementState)
+
+	currentManagementState, err := builder.GetManagementState()
+
+	if err != nil {
+		glog.V(100).Infof("Failed to get current imageRegistry operator management state value due to %s",
+			err.Error())
+
+		return err
+	}
+
+	if currentManagementState != expectedManagementState {
+		glog.V(100).Infof("The current imageRegistry %s ManagementState is %v; it needs to be changed to the %v",
+			builder.Definition.Name, currentManagementState, expectedManagementState)
+
+		builder, err := builder.WithManagementState(expectedManagementState).Update()
+
+		if err != nil {
+			glog.V(100).Infof("Failed to make change to the imageRegistry operator managementState due to %s",
+				err.Error())
+
+			return err
+		}
+
+		newManagementState, err := builder.GetManagementState()
+
+		if err != nil {
+			glog.V(100).Infof("Failed to get current imageRegistry operator managementState value due to %s",
+				err.Error())
+
+			return err
+		}
+
+		if newManagementState != expectedManagementState {
+			return fmt.Errorf("failed to change imageRegistry operator managementState value;"+
+				"expected %v, current value is %v", expectedManagementState, newManagementState)
+		}
+
+		return nil
+	}
+
+	return err
+}
+
+// validate will check that the builder and builder definition are properly initialized before
+// accessing any member fields.
+func (builder *Builder) validate() (bool, error) {
+	resourceCRD := "Config"
```

Comment:
```suggestion
	resourceCRD := "ImageRegistry.Config"
```

Reply:
why?
the object type is "config", there is no such object "ImageRegistry.Config"
(for oc cli it can be config or a full version: "configs.imageregistry.operator.openshift.io"

Reply:
Ahh I see. Can you make it Config.ImageRegistry then? Config is very general so it may not help much in a log message.

Reply:
there is no such resource "Config.ImageRegistry", the "Config" only, this is the name for this resource.
If you don't like it to stay "Config", I can change it to the long version "configs.imageregistry.operator.openshift.io"

Reply:
ah, ok, I see, it will work if it will be "Configs.ImageRegistry".
It should be ok. I'll change it, we'll do it a try (I cannot test it now unfortunately)